lib/semian/protected_resource.rb

```diff
@@ -2,35 +2,59 @@ module Semian
   class ProtectedResource
     extend Forwardable
 
-    def_delegators :@resource, :destroy, :count, :semid, :tickets, :name
+    def_delegators :@bulkhead, :destroy, :count, :semid, :tickets, :name
     def_delegators :@circuit_breaker, :reset, :mark_failed, :mark_success, :request_allowed?,
                    :open?, :closed?, :half_open?
 
-    def initialize(resource, circuit_breaker)
-      @resource = resource
+    attr_reader :bulkhead, :circuit_breaker
```

Comment:
I am not _super_ happy with this not sure how to expose and support the nil assertions. Thoughts?